drools-ansible-rulebook-integration-api/src/main/java/org/drools/ansible/rulebook/integration/api/rulesengine/AbstractRulesEvaluator.java

```diff
@@ -61,7 +63,22 @@ public void startAutomaticPseudoClock(long period, TimeUnit unit) {
         if (log.isInfoEnabled()) {
             log.info("Start automatic pseudo clock with a tick every " + period + " " + unit.toString().toLowerCase());
         }
-        this.automaticClock = new AutomaticPseudoClock(this, period, unit);
+        String automaticPseudoClockCustomClassName = System.getProperty(AUTOMATIC_PSEUDO_CLOCK_CUSTOM_CLASS_NAME_PROPERTY);
+        if (automaticPseudoClockCustomClassName == null) {
+            this.automaticClock = new AutomaticPseudoClock(this, period, unit);
+        } else {
+            // Test purpose only
+            this.automaticClock = instantiateCustomAutomaticPseudoClock(period, unit, automaticPseudoClockCustomClassName);
+        }
```

Comment:
Probably we could achieve the same goal of having an injectable `AutomaticPseudoClock`using a factory? If so I'd prefer it instead of this system property.

Reply:
Sure, I changed it to use a Factory.